test/fake_cloud_firestore_test.dart

```diff
@@ -876,6 +1018,30 @@ void main() {
     }
   });
 
+  test('Should throw Error if data contains nested list', () async {
```

Comment:
Since the two `expectLater`'s do not need to be done in order, it looks like you can safely remove `async` and `await`, and the `test` will wait for everything to complete.

See the examples at https://pub.dev/packages/test#future-matchers.